llvm/test/CodeGen/RISCV/rvv/mscatter-combine.ll

```diff
@@ -116,7 +116,7 @@ define void @stride_one_store(i64 %n, ptr %p) {
 ; RV64:       # %bb.0:
 ; RV64-NEXT:    vsetvli a0, zero, e64, m1, ta, ma
 ; RV64-NEXT:    vmv.v.i v8, 0
-; RV64-NEXT:    vs1r.v v8, (a1)
+; RV64-NEXT:    vse64.v v8, (a1)
```

Comment:
I think this is because we're missing a generic DAG combine from unit strided VP load/store -> regular load/store. I'm not sure if we have any precedent for canonicalising VP ops to non-VP ops already?

Reply:
Do we need to fix this? The semantics is the same, the difference is vs1r doesn't depend on vl while vse64 does.

Reply:
Yeah, I guess the added dependency on vl is a regression but I'm not overly worried about it, since I don't think the loop vectorizer will emit a unit strided masked gather in the first place. But still happy to address it if others want

Reply:
I'm OK with this, maybe just add a FIXME here? The extra complexity may not be worthy to fix this.

Reply:
Not saying we need to fix this before landing, but we do need to fix this.  There's an existing combine for converting an element strided riscv_maked_strided_* into a masked_*.  We should replicate that over the vp.strided form before calling the transition complete.

Reply:
(replying to self to continue a thought)

The needed conversion would be from vp.strided.load to a vp.load not a normal or masked load.  Once that was done, we could in turn conversion to a normal load, but that's a separate step.  The first step is simply getting rid of the "strided" part of the intrinsic.

Reply:
Off the top of my head there's already a generic unit strided VP load - > VP load combine, my original comment should have said "full length AVL and all ones mask"

Reply:
Yeah, you're right.  We're getting to the vse64.v, just not the vs1r.  So it's the second step we're missing.